tests/unit/test_operator.py

```diff
@@ -1,482 +1,398 @@
-    }
-    rel_id = harness.add_relation("db", "mysql_app")
-    harness.add_relation_unit(rel_id, "mysql_app/0")
-    harness.update_relation_data(rel_id, "mysql_app/0", mysql_data)
-
-    default_artifact_root = "not-a-typical-bucket-name"
-    harness.update_config({"default_artifact_root": default_artifact_root})
-
-    # object storage
-    os_data_dict = {
-        "access-key": "minio-access-key",
-        "namespace": "namespace",
-        "port": 1234,
-        "secret-key": "minio-super-secret-key",
-        "secure": True,
-        "service": "service",
-    }
-    os_data = {"_supported_versions": "- v1", "data": yaml.dump(os_data_dict)}
-    os_rel_id = harness.add_relation("object-storage", "storage-provider")
-    harness.add_relation_unit(os_rel_id, "storage-provider/0")
-    harness.update_relation_data(os_rel_id, "storage-provider", os_data)
-
-    # ingress
-    ingress_relation_name = "ingress"
-    relation_version_data = {"_supported_versions": "- v1"}
-    ingress_rel_id = harness.add_relation(
-        ingress_relation_name, f"{ingress_relation_name}-subscriber"
-    )
-    harness.add_relation_unit(ingress_rel_id, f"{ingress_relation_name}-subscriber/0")
+    object_storage_relation_id = harness.add_relation("object-storage", "storage-provider")
+    harness.add_relation_unit(object_storage_relation_id, "storage-provider/0")
     harness.update_relation_data(
-        ingress_rel_id, f"{ingress_relation_name}-subscriber", relation_version_data
-    )
-
-    # Mock away _validate_default_s3_bucket to avoid using boto3/creating clients
-    mocked_validate_default_s3_bucket = mocker.patch("charm.Operator._validate_default_s3_bucket")
-    bucket_name = harness._backend.config_get()["default_artifact_root"]
-    mocked_validate_default_s3_bucket.return_value = bucket_name
-
-    # pod defaults relations setup
-    pod_defaults_rel_name = "pod-defaults"
-    pod_defaults_rel_id = harness.add_relation(
-        "pod-defaults", f"{pod_defaults_rel_name}-subscriber"
-    )
-    harness.add_relation_unit(pod_defaults_rel_id, f"{pod_defaults_rel_name}-subscriber/0")
-
-    harness.begin_with_initial_hooks()
-
-    pod_spec = harness.get_pod_spec()
-    yaml.safe_dump(pod_spec)
-    assert harness.charm.model.unit.status == ActiveStatus()
-
-    charm_name = harness.model.app.name
-    secrets = pod_spec[0]["kubernetesResources"]["secrets"]
-    env_config = pod_spec[0]["containers"][0]["envConfig"]
-    secrets_dict = {s["name"]: s for s in secrets}
-
-    assert (
-        env_config["db-secret"]["secret"]["name"]
-        == secrets_dict[f"{charm_name}-db-secret"]["name"]
-    )
-    assert (
-        b64decode(secrets_dict[f"{charm_name}-db-secret"]["data"]["DB_ROOT_PASSWORD"]).decode(
-            "utf-8"
+        object_storage_relation_id, "storage-provider", object_storage_data
+    )
+    return harness
+
+
+@pytest.fixture(scope="function")
+def relational_db_relation(obejct_storage_relation: Harness) -> Harness:
```

Comment:
This is really `relational_db_relation_with_object_storage_relation` - is there a better way to explain that up front to the reader?